src/lib/components/CodeBlock.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import hljs from 'highlight.js';
+  import 'highlight.js/styles/atom-one-dark.css';
+  import type { Tokens } from 'marked';
+  import { Square2StackIcon } from '@babeard/svelte-heroicons/solid';
+
+  export let token: Tokens.Code;
+
+  let codeElement: HTMLElement;
+
+  function copyToClipboard() {
+    navigator.clipboard.writeText(token.text);
+  }
+
+  onMount(() => {
+    if (codeElement) {
+      hljs.highlightElement(codeElement);
+    }
+  });
+</script>
+
+<div class="relative">
+  <div class="header-bar flex justify-between items-center px-3 py-3">
+    <div class="text-xs">
+      {token.lang || 'plaintext'}
+    </div>
+
+      <button class="flex items-center text-gray-500 hover:text-white p text-xs" on:click={copyToClipboard} aria-label="Copy">
+        <Square2StackIcon class="w-4 h-4 mr-2" aria-hidden="true" />
+        Copy code
+      </button>
+  </div>
+  <pre class="lang-{token.lang}"><code bind:this={codeElement}>{token.text}</code></pre>
+</div>
+
+<style>
```

Comment:
How come we're styling through the `<style>` tag and not with tailwind here? Maybe I'm missing something 🤔